client/lib/performance-tracking/use-performance-tracker-stop.js

```diff
@@ -0,0 +1,26 @@
+/* global requestAnimationFrame:false */
+/**
+ * External dependencies
+ */
+import { useLayoutEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { stop } from '@automattic/browser-data-collector';
+
+/**
+ * Internal dependencies
+ */
+import { getSectionName } from 'state/ui/selectors';
+import config from 'config';
+
+export function usePerformanceTrackerStop() {
+	const sectionName = useSelector( getSectionName );
+
+	// Use `useLayoutEffect` + rAF to be as close as possible to the actual rendering
```

Comment:
I did some tests with Chrome's profiler, performance marks and a few reloads to be sure: in all cases, this will stop the tracking about 10ms before the frame that renders the instrumented components (so `stop` is called ~10 ms earlier) Interestingly, when using two nested `requestAnimationFrame()`, `stop` is called ~100ms late).

I tested both initial page loads and SPA navigations to the instrumented pages.

Reply:
Interesting! I guess you're scheduling your `rAF` before the rendering starts, so it gets called first. If you double-rAF it, you only get the callback after the render has finished, and apparently that render is large and missing its budget, causing a multiple frame delay.